server/stycoon/game.go

```diff
@@ -2,20 +2,272 @@ package stycoon
+		if _, ok := (*resources)[planetId]; !ok {
+			(*resources)[planetId] = make(map[string]*TradingResource)
+		}
+		if _, ok := (*resources)[planetId][resourceIdStr]; !ok {
+			(*resources)[planetId][resourceIdStr] = &TradingResource{}
+		}
+		if buy.Valid {
+			(*resources)[planetId][resourceIdStr].BuyPrice = buy.Float64
+		} else if sell.Valid {
+			(*resources)[planetId][resourceIdStr].SellPrice = sell.Float64
+		}
+	}
+	if err = rows.Err(); err != nil {
+		return fmt.Errorf("rows read failed: %v", err)
+	}
+	return nil
+}
+
+func (game *Game) getTradingResources(objectId int, resources *map[int]map[string]*TradingResource) map[string]TradingResource {
+	planetResources := make(map[string]TradingResource)
+	for key, resource := range (*resources)[objectId] {
+		planetResources[key] = *resource
+	}
+	return planetResources
+}
+
+func (game *Game) getResources(objectId int, resources *map[int]map[string]*TradingResource) map[string]Resource {
+	shipResources := make(map[string]Resource)
+	for key, resource := range (*resources)[objectId] {
+		shipResources[key] = Resource{Amount: resource.Amount}
+	}
+	return shipResources
+}
+
+func (game *Game) getCommodityAmounts() (map[int]map[string]*TradingResource, error) {
+	amounts := make(map[int]map[string]*TradingResource)
+	rows, err := game.db.Query("select `object`, `resource`, `amount` from t_commodity")
+	if err != nil {
+		return amounts, fmt.Errorf("query failed %v", err)
+	}
+	var objectId, resourceId int
+	var amount int64
+	for rows.Next() {
+		err = rows.Scan(&objectId, &resourceId, &amount)
+		if err != nil {
+			return amounts, fmt.Errorf("row read failed %v", err)
+		}
+		if _, ok := amounts[objectId]; !ok {
+			amounts[objectId] = make(map[string]*TradingResource)
+		}
+		amounts[objectId][strconv.Itoa(resourceId)] = &TradingResource{Amount: amount}
+	}
+	if err = rows.Err(); err != nil {
+		return amounts, fmt.Errorf("rows read failed: %v", err)
+	}
+	return amounts, nil
+}
+
+func (game *Game) setPlayers() error {
+	var players = make(map[string]PlayersValue)
+	rows, err := game.db.Query("select `id`, `name`, `color`, `money` from t_player")
+	if err != nil {
+		return fmt.Errorf("query failed %v", err)
+	}
+	var id int
+	var player PlayersValue
+	var color string
+	for rows.Next() {
+		err = rows.Scan(&id, &player.Name, &color, &player.NetWorth.Money)
+		if err != nil {
+			return fmt.Errorf("row read failed %v", err)
+		}
+		err = json.Unmarshal([]byte(color), &player.Color)
+		if err != nil {
+			return err
+		}
+		err = game.setPlayerNetWorth(id, &player.NetWorth)
+		if err != nil {
+			return err
+		}
+		players[strconv.Itoa(id)] = player
+	}
+	if err = rows.Err(); err != nil {
+		return fmt.Errorf("rows read failed: %v", err)
+	}
+	game.players = players
+	return nil
+}
+
+func (game *Game) setPlayerNetWorth(playerId int, playerNetWorth *NetWorth) error {
+	err := game.db.QueryRow("select `price` from v_player_ships_worth where player = ?", playerId).Scan(&playerNetWorth.Ships)
```

Comment:
do not use the views here. instead fetch the data from `t_report_player_score` from the current tick to avoid recalculating all the views repeatedly.
the `t_report_player_score` contains all four columns: commodities, ships, money and total.
use join with t_player to avoid multiple queries.